### ISSUE TITLE
maz-ui Styles in Vue3 messing up all CSS formatting

### ISSUE THREAD

**Issue by gadget-man** [LouisMazel/maz-ui#981]
## Describe the bug

As soon as I try and import the css styles using the @latest version of maz-ui, all the other styles on my website seem to get messed up

## To Reproduce

I'm trying to use the maz-ui MazPhoneNumberInput component in my Vue3 Laravel web app.

I have the following in my main.js:

`import MazPhoneNumberInput from "maz-ui/components/MazPhoneNumberInput";
import 'maz-ui/styles'`

and before I mount the app, I also call app.component('MazPhoneNumberInput', MazPhoneNumberInput);

However, the minute I add the import of the maz-ui styles, my entire css formatting across the site gets messed up. If I inspect an example element in Safari to look for css differences with and without the `import maz-ui/styles` line included, one noticeable difference is that I see a lot of additional css tags appearing. 

In particular, the below item seems to be added, which when disables does solve many of the issues:
*, ::after, ::before {
    border: 0 solid;
    box-sizing: border-box;
}

I'm unable to find where in the maz-ui code this script is included, or how I can override it.

## Expected behavior

I was expecting that 'import 'maz-ui/styles' would manage formatting of my maz-ui component 'MaxPhoneNumberInput', but work alongside my sites other css styles.

## Screenshots
![Screenshot 2024-03-21 at 20 43 14](https://github.com/LouisMazel/maz-ui/assets/24406912/6fbff48f-54ab-4a62-af3d-1b0cba281208)
![Screenshot 2024-03-21 at 20 43 45](https://github.com/LouisMazel/maz-ui/assets/24406912/ea974358-6cdb-4429-b196-ee6ab4b94299)

**Comment by gadget-man** [comment]
> Hi @gadget-man,
> 
> I just released a beta version without the reset CSS. can you test it and tell me if it fixes your CSS style?
> 
> ```shell
> npm install maz-ui@3.36.3-beta.0
> # or npm install maz-ui@beta
> ```

Wow, that was quick! Thanks, that does seem to resolve most of the issues. However, I've noticed that with the latest beta, there is now a grey area above and below the flag, partially obscuring the 'country code' text. (see screenshot).

![Screenshot 2024-03-21 at 21 11 15](https://github.com/LouisMazel/maz-ui/assets/24406912/d1993a5b-a06f-4485-8742-9562c3ccc155)

**Comment by LouisMazel** [comment]
Hmm, it's very strange. 

## For the country flag

Can you share a screenshot of the country flag with the developer console opened with the CSS part? 
I want to find where this gray background color is applied.

Are you on Windows or MacOS?

As it's mentioned [here](https://maz-ui.com/helpers/country-code-to-unicode-flag), the Unicode flag seems to have some bugs on Windows. 
Normally, on the documentation page linked before, you should see [this](https://github.com/LouisMazel/maz-ui/assets/12446546/9a252ecc-9a0b-4270-b89c-e4065c6f6d15)


## For the border

In CSS the good practice is to use this rule to avoid this behavior : 

This CSS rules will remove the default

```css
*, ::before, ::after {
    box-sizing: border-box;
    border-width: 0;
    border-style: solid;
}
```

Test it, disable one by one rules to have the best result

**Comment by gadget-man** [comment]
I'm on Mac Safari. Screenshot enclosed.....
![Screenshot 2024-03-21 at 22 46 22](https://github.com/LouisMazel/maz-ui/assets/24406912/ee653a13-514c-495f-a8b8-7aa901f91352)

**Comment by LouisMazel** [comment]
It's because you don't have any reset CSS. 
So all buttons on your page have the default background color applied by the browser.

You can simply override it with : 

```vue
<template>
  ...
  <MazPhoneNumberInput v-model="phoneNumber" class="maz-phone-number-input" />
  ...
</template>

<style scoped>
  .maz-phone-number-input:deep(.m-phone-number-input__country-flag) {
     background-color: transparent;
  }

  /* For the border issue */
  
  .maz-phone-number-input:deep(*) {
    box-sizing: border-box;
    border-width: 0;
    border-style: solid;
  }
</style>
```

**Comment by gadget-man** [comment]
Thanks that has helped. I was doing similar within maz-ui-variables.scss, which wasn't working. However, it's fine now ive' included it in the <style scoped> as per above.

One final question - I'm passing the data back to the form submission as `cphonenumber` based on the following. However, this returns the number in local format, how do I return the model value in international format?
 `<template> 
    
    <MazPhoneNumberInput
    v-model="cphonenumber"
    name="cphonenumber"
    class="maz-phone-number-input"
    show-code-on-list
    :preferred-countries="['GB', 'BE', 'NL']"
    size="sm"
    @update:model-value = "results = $event"
    />

    <code>
        {{ results }}
    </code>
</template>`

**Comment by LouisMazel** [comment]
I'm not sure I understand

When the phone number is valid, the `v-model` returns the international number formatted  (red arrow)
Otherwise, you have all the data about the phone number in the @update event (blue arrow)

![image](https://github.com/LouisMazel/maz-ui/assets/12446546/8e0aecb5-7c8b-4389-a884-e9c1450c989d)

But I see an error in your code : 

```html
<MazPhoneNumberInput
  v-model="cphonenumber"
  name="cphonenumber"
  class="maz-phone-number-input"
  show-code-on-list
  :preferred-countries="['GB', 'BE', 'NL']"
  size="sm"
  @update="results = $event" <---- Error instead of '@update:model-value' 
/>
```

**Comment by gadget-man** [comment]
I have a Laravel blade which includes the following as the way to initialise the component, and pass any existing saved number down to the child component.
`<phone-number-component id="phone" v-bind:phonenumber="{{ (isset($details)) ? $details['phonenumber'] : '+44' }}"></phone-number-component>

In PhoneNumberComponent.vue, I then have the following:
`
<template> 
    <div>

    <MazPhoneNumberInput
    v-model="cphonenumber"
    name="cphonenumber"
    class="maz-phone-number-input"
    show-code-on-list
    :preferred-countries="['GB', 'BE', 'NL']"
    size="sm"
    @update:model-value = "results = $event"
    />

    <code>
        {{ results }}
    </code>
</div>
</template>

<style scoped>
.maz-phone-number-input:deep(.m-phone-number-input__country-flag) {
  background-color: transparent!important;
}
</style>

<script>
  import { ref } from 'vue'

  export default {
    data: function () {
        return {
            cphonenumber:  '+' + this.phonenumber,
            results: ""
        }

    },
    props: [
        "phonenumber",   ]
};

</script>
`

I can't directly include v-model as 'phonenumber' as I get an error  `v-model cannot be used on a prop, because local prop bindings are not writable.`, so I'm using cphonenumber and updating that with this.phonenumber. (a separate issue is that for some reason the leading '+' isn't being carried forwards so I'm having to re-add it manually. However, when I then submit the form in the Laravel blade, I see from the form submission that cphonenumber is returned, but only using the local number, not the international format.

I suspect I'm doing something really obviously wrong, but I can't see it!

**Comment by gadget-man** [comment]
@LouisMazel  I just updated to latest and the above issue has come back. Was the beta incorporated into updates, and if not is there an alternative way to avoid the reset CSS?

**Comment by jacobg** [comment]
Also having problems with the reset styles destroying co-existence with primevue library. That out-of-date beta version fix doesn't seem like a stable solution. Any other solutions? Thanks.

**Comment by jacobg** [comment]
For now, since I'm only using MazPhoneNumberInput, the following style override is a workaround:
```css
<style lang="scss">
// Use private copied maz ui style with its reset.css manually removed.
// Then we'll add reset.css below only within this component.
// https://github.com/LouisMazel/maz-ui/issues/981
@import '/css/maz-ui.css';

.m-phone-number-input {

  // Apply reset.css only in this component.
  // https://github.com/LouisMazel/maz-ui/blob/master/packages/lib/css/reset.css
  *,
  ::before,
  ::after {
    border-width: 0;
    border-style: solid;
    border-color: currentcolor;
    box-sizing: border-box;
  }

  button {
    background-color: transparent;
  }

  button,
  input,
  optgroup,
  select,
  textarea {
    font-family: inherit;
    font-feature-settings: inherit;
    font-variation-settings: inherit;
    font-size: 100%;
    font-weight: inherit;
    line-height: inherit;
    letter-spacing: inherit;
    color: inherit;
    margin: 0;
  }
}
</style>

```

**Comment by spencerjsmall** [comment]
Also having this issue! Recreating the /css folder locally with everything but reset.css does not work for me. Thus, I can't use the phone input component without messing up my global styles. Help!

**Comment by LouisMazel** [comment]
@jacobg 

CSS reset is required for almost all components.
I can't import the reset css into every component. It will create too much code duplication for people using multiple components.

The CSS reset is generally a good practice (active by default with tailwind for example). But I understand your problem and I will find a solution.

I will make available a new CSS file without CSS reset.
Instead of importing the main.css file, you can import this new CSS file:

```js
import 'maz-ui/styles-no-reset' // or import 'maz-ui/css/main-no-reset.css'
import '@/css/path_to_your_main_file.css'
```

Be careful, UI changes could happen in the components and you will have to fix them yourself.

**Comment by LouisMazel** [comment]
Yes, I want to do it something like this but in the V4. 
I will add a new module to customize the theme with CSS layers directly in JS with a plugin install option and/or a defineTheme method.

After studying the proposal and seeing how primevue works, I reconsidered the solution I gave above.
I will instead add an "m-component" class to all my components and prefix the CSS reset so that it is only active in the components and not globally.

I will also add this CSS reset in a "maz-ui-reset" layer which will make the potential override easier

**Comment by jacobg** [comment]
Awesome @LouisMazel ! Thanks so much!

**Comment by LouisMazel** [comment]
Released in [v3.48.0](https://github.com/LouisMazel/maz-ui/releases/tag/v3.48.0)

**Comment by spencerjsmall** [comment]
@LouisMazel thanks for addressing this! I just installed v3.48.0 and my other components are still being affected by global `tw` styles imported by the library:
<img width="494" alt="Screenshot 2024-11-29 at 11 02 00 AM" src="https://github.com/user-attachments/assets/74010200-ccad-4c71-9d60-afd972158307">
Specifically in the screenshot, ring styles affecting my non-maz Inputs and Buttons are affected.

Am I doing something wrong here? Thanks again for your support.

**Comment by LouisMazel** [comment]
@spencerjsmall 
Are you using Tailwind CSS?

This project is made with Tailwind CSS and these are just the CSS variables it needs to make its utility classes work.
This CSS does not affect your user interface in any way since it is just variables.

**Comment by spencerjsmall** [comment]
@LouisMazel Yes, I'm using tailwind v4 through Nuxt UI v3. I see, these tailwind globals must be coming from the maz-ui tailwind config. As a workaround, I've created a custom `maz.css` file with all the `--maz` styles and without the `tw` styles that are included in the package's `main.css`. 

I still think there's something funky going on, given that the global `tw` styles introduced by maz-ui are setting rings and shadows to invisible/0, but I guess this is the challenge of using multiple component libraries!

Thanks again for your quick attention.

**Comment by LouisMazel** [comment]
Because Nuxt UI is bigger than Maz UI, I think I should do something for that 😄

I'll investigate to see if it's possible to rename all CSS variables generated during build.

**Comment by LouisMazel** [comment]
I just released a beta version with the refacto: v3.48.1-beta.0

I would appreciate it and it would be very helpful for me if you could test it on your side, is that possible for you?

```bash
npm install maz-ui@beta
# or npm install maz-ui@v3.48.1-beta.0
```

Thank you !

**Comment by spencerjsmall** [comment]
@LouisMazel that works beautifully! I see the `-maz` prefix now on those global `tw` styles, which solved it. Thanks so much.

**Comment by LouisMazel** [comment]
Awesome!
I will release it in a new patch version in a few days after some testing on my projects 

I will give an update on this issue :)

**Comment by LouisMazel** [comment]
Thank you very much for testing and your help.
It helps me significantly to improve the lib :)

**Comment by LouisMazel** [comment]
Released in [v3.48.1](https://github.com/LouisMazel/maz-ui/releases/tag/v3.48.1)
